crypto/merklearray/merkle_test.go

```diff
@@ -66,109 +75,998 @@ func (a TestRepeatingArray) Length() uint64 {
 	return a.count
 }
 
-func (a TestRepeatingArray) GetHash(pos uint64) (crypto.Digest, error) {
+func (a TestRepeatingArray) Marshal(pos uint64) (crypto.Hashable, error) {
 	if pos >= a.count {
-		return crypto.Digest{}, fmt.Errorf("pos %d larger than length %d", pos, a.count)
+		return nil, fmt.Errorf("pos %d larger than length %d", pos, a.count)
 	}
 
-	return crypto.HashObj(a.item), nil
+	return a.item, nil
 }
 
+const OutOfBoundErrorString = "larger than leaf count"
```

Comment:
You should replace this with `errors.Is(ErrPosOutOfBound)` using the wrapping technique as per https://go.dev/blog/go1.13-errors